### PR TITLE
CGMES SV : export flows for branches and three windings transformers

### PR DESCRIPTION
**Pull request by zamarrenolm**
Signed-off-by: Luma <zamarrenolm@aia.es>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
CGMES SV export output does not contain flows at terminals of transformers and lines.

**What is the new behavior (if this is a feature change)?**
CGMES SV export output will contain flows at terminals of transformers and lines.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:
We must consider if export of flows should be made configurable. Related data will increase significantly the size of the output SV file.

### CODE REVIEW

**Review by MioRtia — COMMENTED** [review]
Minor comment but otherwise, it is okay for me!

**Comment by annetill** [comment]
@MioRtia you ask the feature to be configurable indeed. Is it still what you want?

**Comment by MioRtia** [comment]
Another question: shouldn't we add the flows at switches?

**Comment by MioRtia** [comment]
> @MioRtia you ask the feature to be configurable indeed. Is it still what you want?

I would like to avoid too much configuration on the contrary! Since it is a requirement of several processes, we can export them each time.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java [hunk 1]
```diff
@@ -270,6 +283,10 @@ private static void writePowerFlow(Terminal terminal, String cimNamespace, XMLSt
     }
 
     private static void writePowerFlow(String terminal, double p, double q, String cimNamespace, XMLStreamWriter writer) {
+        // Export only if flow is a number
+        if (Double.isNaN(p) && Double.isNaN(q)) {
```
[the remark this answers is not in the document]
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/StateVariablesExport.java:287]: Should we be able to export a DC loadflow result ? 

I am not sure SV profile is considering that. 

Found definitions for SV profile talk about _A state variable instance file contains all objects required to describe a steady-state solution_ and _describing the state variables of a power flow solution in terms of **complex** voltages and power flows_.

Also, for equipment that only inject reactive power (`StaticVarCompensator`, `ShuntCompensator` with `g == 0`), `p` should be zero.

**Comment by zamarrenolm** [comment]
> Another question: shouldn't we add the flows at switches?

Good question. Flows at switches are not directly available in IIDM Network (a Switch does not have terminals). But a topology analysis could be done inside a voltage level to assign a flow to them, in a similar way to what is done with zero-impedance branches in the LoadFlow results completion. What do you think @marqueslanauja ?

**Comment by MioRtia** [comment]
> > Another question: shouldn't we add the flows at switches?
> 
> Good question. Flows at switches are not directly available in IIDM Network (a Switch does not have terminals). But a topology analysis could be done inside a voltage level to assign a flow to them, in a similar way to what is done with zero-impedance branches in the LoadFlow results completion. What do you think @marqueslanauja ?

Maybe it should be done in another PR if it is a bit complicated, so we can have this one merged and start validation as soon as possible! So the only modification for me should that if p and/or q is NaN, it should be written as 0 and not ignored/written as empty

**Comment by zamarrenolm** [comment]
> > > Another question: shouldn't we add the flows at switches?
> > 
> > 
> > Good question. Flows at switches are not directly available in IIDM Network (a Switch does not have terminals). But a topology analysis could be done inside a voltage level to assign a flow to them, in a similar way to what is done with zero-impedance branches in the LoadFlow results completion. What do you think @marqueslanauja ?
> 
> Maybe it should be done in another PR if it is a bit complicated, so we can have this one merged and start validation as soon as possible! So the only modification for me should be that if p and/or q is NaN, it should be written as 0 and not ignored/written as empty.

I am not sure of writing a zero if the `p/q` values are `NaN`, we would not be respecting the _semantics_ of these attributes. If p/q values are NaN, it means they have been _intentionally left blank_ by an analysis (the analysis says _I am not able to say anything about this terminal_). If we export an explicit value of zero we are altering that. If an analysis has considered a `ShuntCompensator` or a `StaticVarCompensator` in its calculations, it should have set `p` and `q` different from `NaN` (`p = 0` and `q` set to some value).

I agree on separating the flows at switches in a different PR.

**Review by MioRtia — APPROVED** [review]

**Pull request merged**